Examples/src/main/kotlin/com/mapbox/vision/examples/POIActivityKt.kt

```diff
@@ -0,0 +1,262 @@
+import com.mapbox.vision.mobile.core.models.road.RoadDescription
+import com.mapbox.vision.mobile.core.models.world.WorldCoordinate
+import com.mapbox.vision.mobile.core.models.world.WorldDescription
+import kotlin.math.min
+import kotlinx.android.synthetic.main.activity_main.vision_view
+import kotlinx.android.synthetic.main.activity_poi.*
+
+class POIActivityKt : BaseActivity() {
+
+    companion object {
+
+        // POI will start to appear at this distance, starting with transparent and appearing gradually
+        private const val DRAW_LABEL_MIN_DISTANCE_METERS = 400
+
+        // POI will start to appear from transparent to non-transparent during this first meters of showing distance
+        private const val DISTANCE_FOR_ALPHA_APPEAR_METERS = 150
+        private const val LABEL_SIZE_METERS = 8
+        private const val LABEL_ABOVE_GROUND_METERS = 4
+    }
+
+    private var poiList = listOf<POI>()
+
+    private var visionReplayManagerWasInit = false
+
+    // Download session from tutorial and push to device
+    private var sessionPath = "${Environment.getExternalStorageDirectory().absolutePath}/session"
+
+    private val visionEventsListener = object : VisionEventsListener {
+
+        private var cameraCalibrated = false
+        private val paint = Paint()
+        private var bitmapCameraFrame = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
+
+        override fun onAuthorizationStatusUpdated(authorizationStatus: AuthorizationStatus) {}
+
+        override fun onFrameSegmentationUpdated(frameSegmentation: FrameSegmentation) {}
+
+        override fun onFrameDetectionsUpdated(frameDetections: FrameDetections) {}
+
+        override fun onFrameSignClassificationsUpdated(frameSignClassifications: FrameSignClassifications) {}
+
+        override fun onRoadDescriptionUpdated(roadDescription: RoadDescription) {}
+
+        override fun onWorldDescriptionUpdated(worldDescription: WorldDescription) {}
+
+        override fun onCameraUpdated(camera: Camera) {
+            if (camera.calibrationProgress == 1.0f && !cameraCalibrated) {
+                cameraCalibrated = true
+                bitmapCameraFrame = Bitmap.createBitmap(camera.frameWidth, camera.frameHeight, Bitmap.Config.ARGB_8888)
+                runOnUiThread {
+                    camera_calibration_text.visibility = GONE
+                }
+            } else {
+                runOnUiThread {
+                    camera_calibration_text.text = getString(R.string.camera_calibration_progress, (camera.calibrationProgress * 100).toInt())
+                }
+            }
+        }
+
+        override fun onVehicleStateUpdated(vehicleState: VehicleState) {
+            if (cameraCalibrated) {
+                updateAndDrawPOI(vehicleState.geoLocation)
+            }
+        }
+
+        override fun onCountryUpdated(country: Country) {}
+
+        override fun onUpdateCompleted() {}
+
+        private fun updateAndDrawPOI(newVehicleLocation: GeoLocation) {
```

Comment:
I can find where you `update` POI. maybe rename such method?

Reply:
yeah, it is about updating POI state with current vehicle state rather than POI data itself
Probably updatePOIStateAndDraw() will be better

Reply:
fixed